### PR TITLE
Ensure scratch space is large enough to hold disk image

### PR DESCRIPTION
**Pull request by awels**
**What this PR does / why we need it**:
It is possible the source image is fully allocated and in that case the scratch space would not be large enough to hold the disk image. This takes fsOverhead into account when calculating the scratch space size.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Increased size of scratch space to take fs overhead into account.
```

### CODE REVIEW

File: pkg/controller/util.go [hunk 1]
```diff
@@ -170,6 +171,17 @@ func newScratchPersistentVolumeClaimSpec(pvc *corev1.PersistentVolumeClaim, pod
...
+	sizeRequest := scratchPvcSpec.Spec.Resources.Requests[corev1.ResourceStorage]
+	fsOverhead, err := GetFilesystemOverhead(context.TODO(), client, scratchPvcSpec)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get filesystem overhead for scratch PVC")
+	}
+	fsOverheadFloat, _ := strconv.ParseFloat(string(fsOverhead), 64)
+	usableSpaceRaw := util.CalculateOverheadSpace(fsOverheadFloat, sizeRequest.Value())
+
+	scratchPvcSpec.Spec.Resources.Requests[corev1.ResourceStorage] = *resource.NewScaledQuantity(usableSpaceRaw, 0)
+
```
Comment (mhenriks) [inline pkg/controller/util.go:184]: Correct me if I'm wrong but for FileSystem access mode if target storageclass and scratch storagclass are the same you should be safe to use the same size as target?

Comment (awels) [reply pkg/controller/util.go:184]: Right, but there is nothing requiring this, I need to take a look at this again, I was originally only doing this if the target was block, but it is possible scratch and target are both filesystem but different overhead. So I need to calculate the size we are targeting in the target, and then based on that calculate the appriopriate scratch space size.

File: tests/utils/datavolume.go [hunk 2]
```diff
@@ -374,7 +374,7 @@ func NewDataVolumeWithHTTPImportToBlockPV(dataVolumeName string, size string, ht
 					URL: httpURL,
 				},
 			},
-			PVC: &k8sv1.PersistentVolumeClaimSpec{
+			Storage: &cdiv1.StorageSpec{
```
Comment (akalenyu) [inline tests/utils/datavolume.go:377]: This is quite a challenge to all of the tests that use this API (could even hurt their integrity, if they don't intend to test that)
I'd prefer if we don't do it in this PR at least

Comment (awels) [reply tests/utils/datavolume.go:377]: Well part of the issue is the current state is also not working reliably. And it was constantly failing on this PR.

Comment (akalenyu) [reply tests/utils/datavolume.go]: In that case could just change the spec for those tests, but IIUC the main issue was us mutating the pointer

Comment (awels) [reply tests/utils/datavolume.go]: There are 2 issues. 1. The size was not large enough causing a 100% failure rate. Once I made it a little larger, there was the mutating the pointer issue that got fixed with the deepcopy.

File: tests/cloner_test.go [hunk 3]
```diff
@@ -516,7 +516,7 @@ var _ = Describe("all clone tests", func() {
 				Expect(err).ToNot(HaveOccurred())
 
 				volumeMode := v1.PersistentVolumeFilesystem
-				targetDV := utils.NewDataVolumeForImageCloning("target-dv", "1.1Gi", sourcePvc.Namespace, sourcePvc.Name, nil, &volumeMode)
+				targetDV := utils.NewDataVolumeForImageCloning("target-dv", "1.2Gi", sourcePvc.Namespace, sourcePvc.Name, nil, &volumeMode)
```
Comment (akalenyu) [inline tests/cloner_test.go:519]: TBH I don't fully understand this. Ceph CSI will give you back 2Gi whether you do 1.1 or 1.2

**Review by akalenyu — COMMENTED** [review]
/lgtm

**Pull request merged**